microovn/api/types/services.go

```diff
@@ -99,10 +102,103 @@ const (
 	SrvCentral SrvName = "central"
 	// SrvSwitch - string representation of switch service.
 	SrvSwitch SrvName = "switch"
+	// SrvBgp - string representation of FRR's BGP service
+	SrvBgp SrvName = "bgp"
 )
 
 // ServiceNames - slice containing all known SrvName strings.
-var ServiceNames = []SrvName{SrvChassis, SrvCentral, SrvSwitch}
+var ServiceNames = []SrvName{SrvBgp, SrvChassis, SrvCentral, SrvSwitch}
+
+// ExtraServiceConfig - structure containing optional extra configuration for enabling service
+type ExtraServiceConfig struct {
+	BgpConfig *ExtraBgpConfig `json:"bgpConfig,omitempty" yaml:"bgpConfig,omitempty"`
+}
+
+// ExtraBgpConfig holds extra config options that can be used when enabling BGP config
+type ExtraBgpConfig struct {
+	// ExternalConnection is comma separated list of <iface_name>:<ip4_cidr> values. "iface_name"
+	// is a name of the physical interface that provides connectivity to the external network and
+	// "ip4_cidr" is IPv4 address (e.g. 10.0.0.1/24) that should be assigned to a Logical Router
+	// Port connected to the external network
+	ExternalConnection string `json:"ext_iface,omitempty" yaml:"ext_iface,omitempty"`
+	// Vrf is a VRF table ID into which the OVN will leak its routes
+	Vrf string `json:"vrf,omitempty" yaml:"vrf,omitempty"`
+}
```

Comment:
Do you expect that the final implementation of BGP integration in OVN will require us to configure ASNs at some point? Because currently the ASN is only needed once you are configuring the daemon itself (in `frr`), and that's not something MicroOVN is doing. MicroOVN only sets up the OVN BGP redirection, and the rest of the daemon configuration is up to the user.

Reply:
`Asn` was added as an optional config parameter. When specified, MicroOVN will perform iBGP unnumbered configuration, as you suggested.